core/templates/services/utils.service.ts

```diff
@@ -88,19 +84,23 @@ export class UtilsService {
     }
     for (var i = 0; i < aProps.length; i++) {
       var propName = aProps[i];
-      if (!this.isEquivalent(a[propName], b[propName])) {
+      const getKeyValue = (key: string) =>
+        (obj: Record<string, Object>) => obj[key];
+      if (!this.isEquivalent(
+        getKeyValue(propName)(<Record<string, object>>a),
+        getKeyValue(propName)(<Record<string, object>>b))
+      ) {
```

Comment:
Can you please explain why we had to do it this way?

Reply:
The problem I was getting in that case was this: 
_"Element implicitly has an 'any' type because expression of type 'string' can't be used to index type"._

I found the solution to this problem at https://github.com/microsoft/TypeScript/issues/35859 , but in order not to affect the structure of the Object interface I additionally parse the object to a Record<>

Reply:
Could you add a comment in the code to explain this ((you can include the link to the issue).

Reply:
Done!